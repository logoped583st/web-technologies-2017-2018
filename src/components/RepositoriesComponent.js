import React from 'react';
import { connect } from "react-redux";
import { fetchReposRequest } from '../actions/actions'
import { bindActionCreators } from 'redux';


class RepositoriesComponent extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            type: "All_repos",
            language: "All_Language",
        };
        if (this.props.login !== '' && this.props.login !== undefined) {
            this.props.fetchReposRequest(this.props.login)
        }

        this.languageChange = this.languageChange.bind(this);
        this.typeChange = this.typeChange.bind(this);
    }

    languageChange(event) {
        console.log("CHANGE LANGUAGE")
        this.setState({ language: event.target.value })
    }

    typeChange(event) {
        console.log("CHANGETYPE")
        this.setState({ type: event.target.value })
    }

    render() {
        var repositoreidsArr = [];
        var languageMap = new Map();
        console.log(this.state)
        for (const repos in this.props.repos) {
            if (this.props.repos.hasOwnProperty(repos)) {
                repositoreidsArr.push(this.props.repos[repos])
                languageMap.set(this.props.repos[repos].language, this.props.repos[repos].language)
            }
        }

        var languageArr = [];

        languageMap.forEach(item => {
            languageArr.push(item);
        })

        var localState = this.state;
        console.log(languageArr);
        return (
            (this.props.login !== '' && this.props.login !== undefined) && <div className="REPS">

                <select className="Type" onChange={this.typeChange}>
                    <option value="All_repos">All</option>
                    <option value="public">public</option>
                    <option value="private">private</option>
                </select>

                <select className="Language" onChange={this.languageChange}>
                    <option value="All_Language">All</option>
                    {languageArr.map(item => {
                        if (item !== null) {
                            return <option value={item}>{item}</option>
                        }
                    })}
                </select>

                <h2>Repositories</h2>

                {repositoreidsArr.map(function (item) {
                    switch (localState.type) {
                        case "public": {
                            if (item.private === false && (localState.language === "All_Language" || item.language === localState.language)) {
                                return (<div className="orgs_adds ">
                                    <h2>{item.name}</h2>
                                    <h3> {item.html_url}   {item.language}</h3>
                                </div>)
                            }
                            break;
                        }
                        case "private": {
                            if (item.private === true && (localState.language === "All_Language" || item.language === localState.language)) {
                                return (<div className="orgs_adds ">
                                    <h2>{item.name}</h2>
                                    <h3> {item.html_url}   {item.language}</h3>
                                </div>)
                            }
                            break;
                        }
                        default: {
                            if (localState.language === "All_Language" || item.language === localState.language){
                                return (<div className="orgs_adds">
                                    <h2>{item.name}</h2>
                                    <h3> {item.html_url}   {item.language}</h3>
                                </div>)
                            }
                        }
                    }
                })}


            </div>
        );
    }
}
const getState = (state) => {
    console.log(state);
    return {
        repos: state.reduserRepos,
        login: state.reducerUser.login
    };
};

const mapDispatchToProps = dispatch => bindActionCreators({
    fetchReposRequest,
}, dispatch);

export default connect(getState, mapDispatchToProps)(RepositoriesComponent);