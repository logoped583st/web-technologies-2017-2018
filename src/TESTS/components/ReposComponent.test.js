import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson, mountToJson } from 'enzyme-to-json';
import { HashRouter } from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import RepositoriesComponent from '../../components/RepositoriesComponent';

configure({ adapter: new Adapter() });

describe('UserRepositories component test', () => {
    const mockStore = configureStore([thunk]);
    const initialState = {
        reducerUser: {
            login: 'logoped583st'
        },
        reducerRepos: [
            {
                name: 'adgfds',
                html_url: 'http://KAAAAKAA/',
                language: 'C#',
                private: true,
            },
            {
                name: 'logos',
                html_url: 'http://KAAAAKAA/',
                language: 'Java',
                private: false,
            },
        
        ]
    }

    let store;

    beforeEach(() => {
        store = mockStore(initialState);
    })

    it('UserRepositories component rendered succesfully', () => {
        const userRepositories = mount(
            <HashRouter>
                <RepositoriesComponent store={store} />
            </HashRouter>
        );
        expect(mountToJson(userRepositories)).toMatchSnapshot();
    });

    it('Switch filter', () => {
        const userRepositories = mount(
            <HashRouter>
                <RepositoriesComponent store={store} />
            </HashRouter>
        );
        expect(userRepositories.find(".Type").props().value).toBe('All_repos');
        expect(userRepositories.find(".Language").props().value).toBe('All_Language');
        userRepositories.find(".Type").simulate('change', { target: { value: 'private' } });
        userRepositories.find(".Language").simulate('change', { target: { value: 'C#' } });
        expect(userRepositories.find(".Type").props().value).toBe('private');
        expect(userRepositories.find(".Language").props().value).toBe('C#');
        userRepositories.find(".Type").simulate('change', { target: { value: 'public' } });
        expect(userRepositories.find(".Type").props().value).toBe('public');
        userRepositories.find(".Language").simulate('change', { target: { value: 'All_Language' } });
        expect(userRepositories.find(".Language").props().value).toBe('All_Language');
        userRepositories.find(".Type").simulate('change', { target: { value: 'All_repos' } });
        expect(userRepositories.find(".Type").props().value).toBe('All_repos');
        expect(userRepositories.find(".Language").props().value).toBe('All_Language');
    });

    it('UserRepositories rendered without user login', () => {
        const fakeInitialState = {
            reducerUser: {
                login: ''
            },
            reducerRepos: [
                {
                    name: 'asdasd',
                    html_url: 'http://KAAAAKAA/',
                    language: 'Java',
                    private: true,
                },

                
            ]
        }
        const fakeStore = mockStore(fakeInitialState);
        const userRepositories = mount(
            <HashRouter>
                <RepositoriesComponent store={fakeStore} />
            </HashRouter>
        );
        expect(userRepositories.prop('login')).toBe(undefined);
        expect(userRepositories.prop('reducerRepos')).toBe(undefined);
        expect(mountToJson(userRepositories)).toMatchSnapshot();
    })
});