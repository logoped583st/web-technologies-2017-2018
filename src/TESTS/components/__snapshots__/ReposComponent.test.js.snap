// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRepositories component test UserRepositories component rendered succesfully 1`] = `
<HashRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(RepositoriesComponent)
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <RepositoriesComponent
        fetchReposRequest={[Function]}
        login="logoped583st"
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        storeSubscription={
          Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": [Function],
          }
        }
      >
        <div
          className="REPS"
        >
          <select
            className="Type"
            onChange={[Function]}
            value="All_repos"
          >
            <option
              className="All_repos"
              value="All_repos"
            >
              All
            </option>
            <option
              value="public"
            >
              public
            </option>
            <option
              value="private"
            >
              private
            </option>
          </select>
          <select
            className="Language"
            onChange={[Function]}
            value="All_Language"
          >
            <option
              value="All_Language"
            >
              All
            </option>
          </select>
          <h2>
            Repositories
          </h2>
        </div>
      </RepositoriesComponent>
    </Connect(RepositoriesComponent)>
  </Router>
</HashRouter>
`;

exports[`UserRepositories component test UserRepositories rendered without user login 1`] = `
<HashRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(RepositoriesComponent)
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <RepositoriesComponent
        fetchReposRequest={[Function]}
        login=""
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        storeSubscription={
          Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": [Function],
          }
        }
      />
    </Connect(RepositoriesComponent)>
  </Router>
</HashRouter>
`;
